src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java

```diff
@@ -1,31 +1,121 @@
 package com.endava.internship.mocking.service;
 
+import com.endava.internship.mocking.model.Payment;
+import com.endava.internship.mocking.model.Status;
+import com.endava.internship.mocking.model.User;
+import com.endava.internship.mocking.repository.PaymentRepository;
+import com.endava.internship.mocking.repository.UserRepository;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
 import org.mockito.junit.jupiter.MockitoExtension;
 
+import java.util.Optional;
+import java.util.UUID;
+import java.util.List;
+import java.util.Arrays;
+import java.util.ArrayList;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+
 @ExtendWith(MockitoExtension.class)
 class PaymentServiceTest {
 
 
+    @Mock
+    private PaymentRepository paymentRepository;
+
+    @Mock
+    private UserRepository userRepository;
+
+    @Mock
+    private ValidationService validationService;
+
+    @InjectMocks
+    private PaymentService paymentService;
+
+    @Captor
+    ArgumentCaptor<Payment> captor;
+
+    private static User user;
+    private static Payment payment;
+    private static Integer id;
+    private static Double amount;
+
```

Comment:
Add an afterEach method and check verifyNoMoreInteractions for every mock

Reply:
Done